src/TraceEvent/EventSources/ProcessMetadataEventSource.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Tracing
+{
+    [EventSource(Name = "ProcessMetadataEventSource")]
+    public sealed class ProcessMetadataEventSource : EventSource
+    {
+        public class Tasks
+        {
+            public const EventTask Process = (EventTask)1;
+            public const EventTask Thread = (EventTask)2;
+            public const EventTask Module = (EventTask)3;
+        }
+
+        [Event(1, Opcode = EventOpcode.Start, Task = Tasks.Process)]
+        public void ProcessStart(long ProcessId, long ParentProcessId, string Executable, string CommandLine)
+        {
+            this.WriteEvent(1, ProcessId, ParentProcessId, Executable, CommandLine);
+        }
+
+        [Event(2, Opcode = EventOpcode.Stop, Task = Tasks.Process)]
+        public void ProcessExit(long ProcessId, long ParentProcessId, int ExitCode, string Executable, string CommandLine)
+        {
+            this.WriteEvent(2, ProcessId, ParentProcessId, ExitCode, Executable, CommandLine);
+        }
+
+        [Event(3, Opcode = EventOpcode.Start, Task = Tasks.Thread)]
+        public void ThreadCreate(long ProcessId, long ThreadId, ulong StackBaseAddress, string ThreadName)
+        {
+            this.WriteEvent(3, ProcessId, ThreadId, StackBaseAddress, ThreadName);
+        }
+
+        [Event(4, Opcode = EventOpcode.Stop, Task = Tasks.Thread)]
+        public void ThreadDestroy(long ProcessId, long ThreadId, ulong StackBaseAddress, string ThreadName)
+        {
+            this.WriteEvent(4, ProcessId, ThreadId, StackBaseAddress, ThreadName);
+        }
+
+        [Event(5, Opcode = EventOpcode.Start, Task = Tasks.Module)]
+        public void ModuleLoad(long ProcessId, ulong LoadAddress, long ModuleSize, Guid DebugGuid, int DebugAge, string ModuleFilePath, string DebugModuleFileName)
```

Comment:
As luck has it, ELF Debug IDs are 20-bytes. so I contemplated making this a byte[] but then thought about MachO which is also a GUID without any age, so I thought we could encode it as 0. And we could hack into the ELF IDs as guid + age to take the 20 bytes.

I just picked this and not a byte array because it was more convenient. @noahfalk thoughts?

Reply:
Gotcha - yes, interested in @noahfalk's thoughts here.  Do we have any other precedent?

Reply:
I'd be tempted to encode it as a byte[] or string. Fixed size types would be a little more performant but given these shouldn't be frequent events I'd favor an encoding that feels straightforward. 

If the goal is to be able to look up the images and symbols on a symbol server then the data we need is:
format - PE/ELF/MachO
  - PE - filesize, timestamp, filename, debug signature, age, and the codeview debug directory major/minor version (major=0x100,minor=504D is a sentinel value that indicates portable pdb is supported - [PE spec](https://github.com/tmat/corefx/blob/master/src/System.Reflection.Metadata/specs/PE-COFF.md)). 
  - ELF - build-id and filename
  - MachO - uuid and name

[SSQP spec](https://github.com/dotnet/symstore/blob/master/docs/specs/SSQP_Key_Conventions.md)